### PR TITLE
Issue #SB-SH-120 feat: Changes to course progress calculation

### PR DESCRIPTION
**Pull request by rahulshukla**
Changes to course progress calculation for videos, youtube content & pdf

### CODE REVIEW

File: player/public/js/baseLauncher.js [hunk 1]
```diff
@@ -128,8 +130,12 @@ org.ekstep.contentrenderer.baseLauncher = Class.extend({
 		if (this.sleepMode) return
 		if (TelemetryService.instance && TelemetryService.instance.telemetryStartActive()) {
 			var telemetryEndData = {}
+			var endpageSeen = [{ "endpageseen" : this.isEndPageSeen }]
+			var contentProgress = [{ "progress" : this.contentProgress() }]
+			var contentSummary = (_.isFunction(this.contentPlaySummary)) ? this.contentPlaySummary() : ''
```
Comment (gouravmore) [inline player/public/js/baseLauncher.js:135]: @rahulshukla I think the `contentPlaySummary()` method should define in the base class

Comment (rahulshukla) [reply player/public/js/baseLauncher.js:135]: @gouravmore  What do you mean by base class here and why?

Comment (vinukumar-vs) [inline player/public/js/baseLauncher.js:135]: Define the interface methods for each of the common summary properties, similar to contentProgress. Each launcher has to override/implement these methods.

Then addition summary specific to contentType will be given by another interface method additionalContentSummary().

Comment (vinukumar-vs) [reply player/public/js/baseLauncher.js:135]: Base launcher should create common summary event. So that for any content, these fields will exist.

**Comment by vinukumar-vs** [comment]
@rahulshukla 
Create the JIRA issue to refactor the code using the interface class to create summary object.

**Pull request merged**